app/controllers/unattended_controller.rb

```diff
@@ -141,41 +142,19 @@ def get_built_host_details
     @host = Foreman::UnattendedInstallation::HostFinder.new(query_params: query_params).search
   end
 
-  def verify_valid_host_token
-    return true unless @host&.token_expired?
-    render_custom_error(
-      :precondition_failed,
-      N_('%{controller}: provisioning token for host %{host} expired'),
-      { :controller => controller_name, :host => @host.name }
-    )
-    false
-  end
-
   def verify_found_host
-    return false if host_not_found?(@host) || host_os_is_missing?(@host) || host_os_family_is_missing?(@host)
-    logger.debug "Found #{@host}"
-    true
-  end
+    host_verifier = Foreman::UnattendedInstallation::HostVerifier.new(@host, request_ip: request.env['REMOTE_ADDR'],
+                                                                             for_host_template: (action_name == 'host_template'))
```

Comment:
Being a huge fan of composition over inheritance/concerns, I am good with this. Probably `UnattendedRequestValidator` would be better name.

Reply:
Thought about this one and I was almost going to agree with you however since it is very host specific and not exactly request specific we should keep it as HostVerifier. As above, this is namespace'd to `UnattendedInstallation` so its implying that it verifies the host for an unattended installation.